test/test_encryption.py

```diff
@@ -578,120 +598,128 @@ def create_key_vault(vault, *data_keys):
         super(TestDataKeyDoubleEncryption, cls).setUpClass()
-
-    @staticmethod
-    def kms_providers():
-        return {'aws': AWS_CREDS, 'local': {'key': LOCAL_MASTER_KEY}}
-
-    def test_data_key(self):
-        listener = OvertCommandListener()
-        client = rs_or_single_client(event_listeners=[listener])
-        self.addCleanup(client.close)
-        client.db.coll.drop()
-        vault = create_key_vault(client.keyvault.datakeys)
-        self.addCleanup(vault.drop)
+        cls.listener = OvertCommandListener()
+        cls.client = rs_or_single_client(event_listeners=[cls.listener])
+        cls.client.db.coll.drop()
+        cls.vault = create_key_vault(cls.client.keyvault.datakeys)
 
         # Configure the encrypted field via the local schema_map option.
         schemas = {
-          "db.coll": {
-            "bsonType": "object",
-            "properties": {
-              "encrypted_placeholder": {
-                "encrypt": {
-                  "keyId": "/placeholder",
-                  "bsonType": "string",
-                  "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Random"
+            "db.coll": {
+                "bsonType": "object",
+                "properties": {
+                    "encrypted_placeholder": {
+                        "encrypt": {
+                            "keyId": "/placeholder",
+                            "bsonType": "string",
+                            "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Random"
+                        }
+                    }
                 }
-              }
             }
-          }
         }
         opts = AutoEncryptionOpts(
-            self.kms_providers(), 'keyvault.datakeys', schema_map=schemas)
-        client_encrypted = rs_or_single_client(
+            cls.KMS_PROVIDERS, 'keyvault.datakeys', schema_map=schemas)
+        cls.client_encrypted = rs_or_single_client(
             auto_encryption_opts=opts, uuidRepresentation='standard')
-        self.addCleanup(client_encrypted.close)
-
-        client_encryption = ClientEncryption(
-            self.kms_providers(), 'keyvault.datakeys', client, OPTS)
-        self.addCleanup(client_encryption.close)
+        cls.client_encryption = ClientEncryption(
+            cls.KMS_PROVIDERS, 'keyvault.datakeys', cls.client, OPTS)
 
-        # Local create data key.
-        listener.reset()
-        local_datakey_id = client_encryption.create_data_key(
-            'local', key_alt_names=['local_altname'])
-        self.assertBinaryUUID(local_datakey_id)
-        cmd = listener.results['started'][-1]
-        self.assertEqual('insert', cmd.command_name)
-        self.assertEqual({'w': 'majority'}, cmd.command.get('writeConcern'))
-        docs = list(vault.find({'_id': local_datakey_id}))
-        self.assertEqual(len(docs), 1)
-        self.assertEqual(docs[0]['masterKey']['provider'], 'local')
+    @classmethod
+    def tearDownClass(cls):
+        cls.client.close()
+        cls.vault.drop()
```

Comment:
These two lines should be swapped. Dropping the vault collection reopens the client we just closed.